baseapp/options.go

```diff
@@ -216,7 +216,7 @@ func (app *BaseApp) SetSnapshot(snapshotStore *snapshots.Store, opts snapshottyp
 	if app.sealed {
 		panic("SetSnapshot() on sealed BaseApp")
 	}
-	if snapshotStore == nil || opts.Interval == snapshottypes.SnapshotIntervalOff {
+	if snapshotStore == nil {
```

Comment:
If we remove this, then `SnapshotIntervalOff` is never used anywhere. Maybe we can remove this `const` as well.

When `opts.Interval` is not set / 0, everything will be pruned.
https://github.com/cosmos/cosmos-sdk/blob/04db139e8ab6c6b8aade44898aa1bbcda6b24207/store/pruning/manager.go#L138-L143